include/bitcoin/node/parser.hpp

```diff
@@ -53,6 +54,17 @@ class BCN_API parser
 
     /// The populated configuration settings values.
     configuration configured;
+
+private:
+    template <typename T>
+    const boost::program_options::value_semantic* property(
+        void (bc::settings::* method)(T))
+    {
+        return boost::program_options::value<T>()->notifier([&, method](T value)
+        {
+            (configured.bitcoin.*method)(value);
+        });
+    }
```

Comment:
You should move this to `bc::config::parser`, although since it's not fully generalized:
```cpp
(bc::settings::* method)
...
(configured.bitcoin.*method)
```
that may be a challenge.

Reply:
`configured` is defined here. I think that it's possible to move the template to `bc::config::parser` but we'd need to add a second parameter to pass in `configured.bitcoin`
```cpp
const boost::program_options::value_semantic* property(
        void (bc::settings::* method)(T), bc::settings& bitcoin_settings)
```
.

Reply:
Or you could use a macro, which is much simpler and sufficiently type safe :).

Reply:
:laughing:

Reply:
Done.